### PR TITLE
RET-3715 Orders and request sub-page statuses

### PR DESCRIPTION
**Pull request by brendanmchugh**
### JIRA link
https://tools.hmcts.net/jira/browse/RET-3715

### Change description

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/controllers/TribunalOrderOrRequestDetailsController.ts [hunk 1]
```diff
@@ -35,6 +35,10 @@ export default class TribunalOrderOrRequestDetailsController {
     const respondButton = !selectedRequestOrOrder.value.respondCollection?.some(
       r => r.value.from === Applicant.CLAIMANT
     );
+    const responseRequired =
```
Comment (robin-ieong) [inline src/main/controllers/TribunalOrderOrRequestDetailsController.ts:38]: Could this logic here be unit tested? might wanna have it as a function in a separate class as well

File: src/test/unit/controller/helpers/TribunalOrderOrRequestHelper.test.ts [hunk 2]
```diff
@@ -144,9 +149,38 @@ describe('Tribunal order or request helper', () => {
...
+    const populatedNotification = populateNotificationsWithRedirectLinksAndStatusColors(
+      [mockNotificationResponseReq],
+      'url',
+      translations
+    )[0];
+    expect(populatedNotification.statusColor).toEqual('--red');
     expect(populatedNotification.displayStatus).toEqual('Not started yet');
   });
 
+  it('hould populate notification with correct status when not required to respond', () => {
```
Comment (robin-ieong) [inline src/test/unit/controller/helpers/TribunalOrderOrRequestHelper.test.ts:164]: think we're missing test for "Submitted" status here?

File: src/main/controllers/TribunalOrderOrRequestDetailsController.ts (hunk 1 above)
Comment (brendanmchugh) [reply src/main/controllers/TribunalOrderOrRequestDetailsController.ts:38]: This responseRequired variable check is to toggle the display of a paragraph which states "You must respond to the tribunal". Waiting on confirmation that this is okay or not, it wasn't part of the ticket.

Comment (brendanmchugh) [reply src/main/controllers/TribunalOrderOrRequestDetailsController.ts:38]: Added a unit test for this controller

**Review by robin-ieong — APPROVED** [review]

**Review by lvo700 — DISMISSED** [review]

File: src/test/unit/controller/TribunalOrderOrRequestDetailsController.test.ts [hunk 3]
```diff
@@ -34,4 +34,63 @@ describe('Respondent order or request details Controller', () => {
     controller.get(request, response);
     expect(response.render).toBeDefined();
   });
+  it('should pass a value to indicate whether a response is required when true', async () => {
```
Comment (Itay-Kirsh) [inline src/test/unit/controller/TribunalOrderOrRequestDetailsController.test.ts:37]: Add spaces between test cases.

File: src/test/unit/controller/helpers/TribunalOrderOrRequestHelper.test.ts [hunk 4]
```diff
@@ -144,9 +150,48 @@ describe('Tribunal order or request helper', () => {
       '/tribunal-order-or-request-details/2c6ae9f6-66cd-4a6b-86fa-0eabcb64bf28?lng=en'
     );
     expect(populatedNotification.statusColor).toEqual('--red');
+    expect(populatedNotification.displayStatus).toEqual('Not viewed yet');
+  });
+  it('should populate notification with correct status when required to respond and no response exists', () => {
```
Comment (Itay-Kirsh) [inline src/test/unit/controller/helpers/TribunalOrderOrRequestHelper.test.ts:155]: add spaces here too

**Review by robin-ieong — APPROVED** [review]

**Pull request merged**